app/models/form_profiles/va21686c.rb

```diff
@@ -1,11 +1,249 @@
+  end
+
+  def detect_file_number(file_number)
+    return if file_number.nil? || file_number.match(/^\d{3}-\d{2}-\d{4}$/)
+    file_number
+  end
+
+  def prefill_marriage(marriage)
+    return unless marriage
+    VA21686c::FormMarriage.new(
+      {
+        date_of_marriage: convert_date(marriage['marriageDate']), # ???
+        location_of_marriage: prefill_location(marriage['country'], marriage['city'], marriage['state']),
+        spouse_full_name: prefill_name(marriage),
+        reason_for_separation: marriage['marriageTerminationReasonType'],
+        date_of_separation: convert_date(marriage['terminatedDate']),
+        location_of_separation: prefill_location(marriage['endCountry'], marriage['endCity'], marriage['endState'])
+      }.compact
+    )
+  end
+
+  def prefill_location(country, city, state)
+    country ||= {}
+    VA21686c::FormLocation.new(
+      {
+        country_dropdown: country['dropDownCountry'],
+        country_text: country['textCountry'],
+        city: city,
+        state: state
+      }.compact
+    )
+  end
+
+  def convert_date(date)
+    return unless date
+    Date.strptime(date.to_s, '%Q').strftime('%Y-%m-%d')
+  end
+
+  def prefill_dependents(children)
+    return [] if children.blank?
+    children.map do |child|
+      VA21686c::FormDependent.new(
+        {
+          full_name: prefill_name(child),
+          child_date_of_birth: convert_date(child['dateOfBirth']),
+          child_in_household: child['sameResidency'],
+          child_address: prefill_address(child['address']),
+          child_social_security_number: convert_ssn(child['ssn']),
+          attending_college: child['attendedSchool'],
+          disabled: child['disabled'],
+          married: child['married'],
+          child_place_of_birth: prefill_location(child['countryOfBirth'], child['cityOfBirth'], child['stateOfBirth']),
+          child_relationship_type: child['childRelationshipType']
+        }.compact
+      )
+    end
+  end
+
+  def prefill_name(person)
+    VA21686c::FormFullName.new(
+      {
+        first: person['firstName'],
+        last: person['lastName'],
+        middle: person['middleName']
+      }.compact
+    )
+  end
+
+  def prefill_address(address)
+    return unless address
+    address = VA21686c::FormAddress.new(
+      {
+        address_type: address['addressLocality'],
+        street: address['addressLine1'],
+        street2: address['addressLine2'],
+        street3: address['addressLine3'],
+        city: address['city'],
+        state: address['state'],
+        postal_code: "#{address['zipCode']}#{"-#{address['zipLastFour']}" if address['zipLastFour'].present?}",
+        country: address.dig('country', 'dropDownCountry'),
+        country_text: address.dig('country', 'textCountry'),
+        post_office: address['postOffice'],
+        postal_type: address['postalType']
+      }.compact
+    )
+    return if address.attributes.values.uniq.all? { |x| ['DOMESTIC', ''].include? x }
```

Comment:
`retrieve` returns blank addresses with all empty attributes except `addressLocality` will be `DOMESTIC`